### ISSUE TITLE
Issue with aka ms getnav

### ISSUE THREAD

**Issue by ajansari** [Microsoft/nav-docker#202]
Hi,

I'm trying to set up a new NAV 2018 image / vm on Azure using aka.ms/getnav, and it keeps getting stuck on the same error (tried twice so far): 

_The property 'Tenants' cannot be found on this object. Verify that the property exists._

The NAV Docker Image I specified in the template / form was:

_microsoft/dynamics-nav:2018-na_

And while filling the template / form, I set "Click Once" to Yes, and put in O365 username and password credentials.

Any ideas what may be causing this?

Thanks, AJ

**Comment by freddydk** [comment]
Would you happen to have a VM which you can email me the url/credentials to - then I will have a look.

**Comment by freddydk** [comment]
Just setup a machine and my log file looks like this:
Could you share where you see the error and what happens?

8:05:10 AM Ready for connections!
8:05:10 AM Initialization took 271 seconds
8:05:10 AM 
8:05:10 AM http://fkna.westeurope.cloudapp.azure.com:8080/al-0.12.20065.vsix
8:05:10 AM Files:
8:05:10 AM 
8:05:10 AM ClickOnce Manifest : http://fkna.westeurope.cloudapp.azure.com:8080/NAV
8:05:10 AM Dev. ServerInstance : NAV
8:05:10 AM Dev. Server : https://fkna.westeurope.cloudapp.azure.com
8:05:10 AM Web Client : https://fkna.westeurope.cloudapp.azure.com/NAV/
8:05:10 AM Container Dns Name : fkna.westeurope.cloudapp.azure.com
8:05:10 AM Container Hostname : navserver
8:05:10 AM Container IP Address: 172.17.189.169
8:05:10 AM Creating ClickOnce Manifest
8:05:10 AM Creating SUPER user
8:05:10 AM Creating admin as SQL User and add to sysadmin
8:05:10 AM Setting SA Password and enabling SA
8:05:10 AM Creating Host Windows user
8:05:10 AM Creating Windows user admin
8:05:10 AM Creating http download site
8:05:10 AM Creating DotNetCore Web Server Instance
8:05:10 AM Starting NAV Service Tier
8:05:10 AM Changing Server config to NavUserPassword to enable basic web services
8:05:09 AM Modifying Service Tier Config File with Instance Specific Settings
8:05:09 AM Import Certificate to LocalMachine\my
8:05:09 AM Certificate File Thumbprint D8365FE9BF75415C38D8F1E8717E5982ADA4BE98
8:05:09 AM Starting Internet Information Server
8:05:09 AM Starting Local SQL Server
8:05:09 AM Using AccessControlService Authentication
8:05:09 AM PublicDnsName is fkna.westeurope.cloudapp.azure.com
8:05:09 AM Hostname is navserver
8:05:09 AM Starting Container
8:05:09 AM Initializing...
8:05:09 AM Container output
8:03:00 AM Creating Aad Apps for Office 365 integration
8:02:57 AM Copying .vsix and Certificate to C:\ProgramData\navcontainerhelper\Extensions\navserver
7:57:30 AM Running microsoft/dynamics-nav:2018-na (this will take a few minutes)
7:57:30 AM Locale en-US
7:57:30 AM Version 11.0.21836.0
7:57:30 AM Country na
7:57:30 AM Using image microsoft/dynamics-nav:2018-na
7:57:26 AM Installing OpenXML 2.5
7:57:23 AM Installing SQL Native Client
7:57:08 AM Installing Visual C++ Redist
7:07:01 AM Pulling microsoft/dynamics-nav:2018-na (this might take ~30 minutes)
7:07:01 AM Adding Landing Page to Startup Group
7:07:00 AM Add Import navcontainerhelper to PowerShell profile
7:07:00 AM Disabling Server Manager Open At Logon
7:07:00 AM Enabling Font Download in IE
7:07:00 AM Enabling File Download in IE
7:02:22 AM Launching SetupVm
7:02:21 AM Starting docker
7:00:26 AM Restarting computer and start Installation tasks
7:00:11 AM Enabling Docker API
6:59:10 AM Installing Docker
6:59:06 AM Requesting certificate
6:58:03 AM Performing Lets Encrypt challenge to default web site
6:58:01 AM Creating new dns Identifier
6:57:59 AM Register Contact EMail address and accept Terms Of Service
6:57:58 AM Initializing ACMEVault
6:57:32 AM Installing ACMESharp PowerShell modules
6:57:32 AM Using Lets Encrypt certificate
6:57:16 AM Install Nav Container Helper from PowerShell Gallery
6:57:15 AM Downloading C:\DEMO\Install-VS2017Community.ps1
6:57:15 AM Downloading c:\demo\SetupStart.ps1
6:57:15 AM Downloading c:\demo\SetupVm.ps1
6:57:15 AM Downloading c:\demo\SetupNavContainer.ps1
6:57:15 AM Downloading c:\demo\SetupDesktop.ps1
6:57:15 AM Turning off IE Enhanced Security Configuration
6:57:15 AM Creating Connect.rdp
6:57:14 AM Downloading C:\inetpub\wwwroot\web.config
6:57:14 AM Downloading C:\inetpub\wwwroot\Microsoft.png
6:57:14 AM Downloading C:\inetpub\wwwroot\line.png
6:57:14 AM Downloading C:\inetpub\wwwroot\status.aspx
6:57:14 AM Downloading C:\inetpub\wwwroot\default.aspx
6:53:55 AM Installing Internet Information Server (this might take a few minutes)
6:53:50 AM Installing NuGet Package Provider
6:53:27 AM TemplateLink: https://raw.githubusercontent.com/Microsoft/nav-arm-templates/master/getnav.json
6:53:27 AM Starting initialization

**Comment by ajansari** [comment]
Hi Freddy,

I'll e-mail you the VM and login information shortly. Need to change passwords. :-)

Here is the log below. It crashes when creating Aad Apps for Office 365 integration. In this try yesterday, I skipped the LetsEncrypt option. But earlier in the day yesterday, I had chosen that as well, and it had failed at exactly the same point.
=====

3:53:19 AM The property 'Tenants' cannot be found on this object. Verify that the property exists.
3:51:38 AM Creating Aad Apps for Office 365 integration
3:51:36 AM Copying .vsix and Certificate to C:\ProgramData\navcontainerhelper\Extensions\navserver
3:47:11 AM Running microsoft/dynamics-nav:2018-na (this will take a few minutes)
3:47:10 AM Locale en-US
3:47:10 AM Version 11.0.21836.0
3:47:10 AM Country na
3:47:10 AM Using image microsoft/dynamics-nav:2018-na
3:47:06 AM Installing OpenXML 2.5
3:47:04 AM Installing SQL Native Client
3:46:53 AM Installing Visual C++ Redist
3:06:11 AM Pulling microsoft/dynamics-nav:2018-na (this might take ~30 minutes)
3:06:10 AM Adding Landing Page to Startup Group
3:06:10 AM Add Import navcontainerhelper to PowerShell profile
3:06:10 AM Disabling Server Manager Open At Logon
3:06:10 AM Enabling Font Download in IE
3:06:10 AM Enabling File Download in IE
3:03:38 AM Launching SetupVm
3:03:37 AM Starting docker
3:01:35 AM Restarting computer and start Installation tasks
3:01:32 AM Enabling Docker API
3:00:49 AM Installing Docker
3:00:37 AM Install Nav Container Helper from PowerShell Gallery
3:00:37 AM Downloading C:\DEMO\Install-VS2017Community.ps1
3:00:37 AM Downloading c:\demo\SetupStart.ps1
3:00:37 AM Downloading c:\demo\SetupVm.ps1
3:00:37 AM Downloading c:\demo\SetupNavContainer.ps1
3:00:37 AM Downloading c:\demo\SetupDesktop.ps1
3:00:37 AM Turning off IE Enhanced Security Configuration
3:00:37 AM Creating Connect.rdp
3:00:36 AM Downloading C:\inetpub\wwwroot\web.config
3:00:36 AM Downloading C:\inetpub\wwwroot\Microsoft.png
3:00:36 AM Downloading C:\inetpub\wwwroot\line.png
3:00:36 AM Downloading C:\inetpub\wwwroot\status.aspx
3:00:36 AM Downloading C:\inetpub\wwwroot\default.aspx
2:58:18 AM Installing Internet Information Server (this might take a few minutes)
2:58:09 AM Installing NuGet Package Provider
2:57:50 AM TemplateLink: https://raw.githubusercontent.com/Microsoft/nav-arm-templates/master/getnav.json
2:57:50 AM Starting initialization

**Comment by freddydk** [comment]
This bug requires a new version of NavContainerHelper - will be fixed later today or tomorrow.

**Comment by AlphaGeeKUK** [comment]
Great, thanks Freddy.

Sent from my ZX Spectrum

________________________________

**Comment by freddydk** [comment]
If you login to your VM after you get this error, open NavContainerhelper powershell ISE and run replace-navservercontainer - then it will probably fix the machine (if I am right)

**Comment by AlphaGeeKUK** [comment]
I'll try in about an hour from now and let you know.

Sent from my ZX Spectrum

________________________________

**Comment by freddydk** [comment]
The Replace-NavServerContainer has always fixed this - it seems to be a timing issue.
I have never seen this problem myself.
I am creating a new version of NavContainerHelper, which will insert a delay where the problem seems to be and hope it will fix it for you guys as well.

**Comment by freddydk** [comment]
aka.ms/getnav is now updated and you should not get the Tenants error anymore.
If you could try and give me a heads up, that would be really nice.
Thanks

**Comment by ajansari** [comment]
Thanks Freddy! I'll try this in a bit and report my results.

**Comment by ajansari** [comment]
A couple hours ago, I logged into the Azure VM that I'd sent you credentials for, and I see that you had been working on some scripts in PowerShell. I tried the Replace-NavServerContainer from PowerShell ISE but that returned an error (which I absentmindedly forgot to note down).

Since you noted that aka.ms/getnav had been updated, I decided to start from scratch (the image I'm pulling is microsoft/dynamics-nav:2018-na). I have Click Once enabled, Office365 credentials populated, Lets Encrypt Cert option enabled with a proper e-mail address (same as the O365 e-mail). Unfortunately, it failed at exactly the same point.

Right now, I'm running Replace-NavServerContainer in PowerShell ISE. Will send you an update when it's finished.

xxxxxx

View Landing Page   Disable refresh 

1:20:26 AM The property 'Tenants' cannot be found on this object. Verify that the property exists.
1:18:33 AM Creating Aad Apps for Office 365 integration
1:18:32 AM Copying .vsix and Certificate to C:\ProgramData\navcontainerhelper\Extensions\navserver
1:13:53 AM Running microsoft/dynamics-nav:2018-na (this will take a few minutes)
1:13:53 AM Locale en-US
1:13:53 AM Version 11.0.22292.0
1:13:53 AM Country na
1:13:53 AM Using image microsoft/dynamics-nav:2018-na
1:13:50 AM Installing OpenXML 2.5
1:13:47 AM Installing SQL Native Client
1:13:36 AM Installing Visual C++ Redist
12:29:23 AM Pulling microsoft/dynamics-nav:2018-na (this might take ~30 minutes)
12:28:24 AM Adding Landing Page to Startup Group
12:28:23 AM Add Import navcontainerhelper to PowerShell profile
12:28:23 AM Disabling Server Manager Open At Logon
12:28:23 AM Enabling Font Download in IE
12:28:22 AM Enabling File Download in IE
12:27:40 AM Launching SetupVm
12:27:39 AM Starting docker
12:25:58 AM Restarting computer and start Installation tasks
12:25:54 AM Enabling Docker API
12:24:38 AM Installing Docker
12:24:31 AM Requesting certificate
12:23:28 AM Performing Lets Encrypt challenge to default web site
12:23:27 AM Creating new dns Identifier
12:23:25 AM Register Contact EMail address and accept Terms Of Service
12:23:23 AM Initializing ACMEVault
12:23:03 AM Installing ACMESharp PowerShell modules
12:23:03 AM Using Lets Encrypt certificate
12:22:40 AM Install Nav Container Helper from PowerShell Gallery
12:22:40 AM Downloading C:\DEMO\Install-VS2017Community.ps1
12:22:40 AM Downloading c:\demo\SetupStart.ps1
12:22:40 AM Downloading c:\demo\SetupVm.ps1
12:22:39 AM Downloading c:\demo\SetupNavContainer.ps1
12:22:39 AM Downloading c:\demo\SetupDesktop.ps1
12:22:39 AM Turning off IE Enhanced Security Configuration
12:22:39 AM Creating Connect.rdp
12:22:39 AM Downloading C:\inetpub\wwwroot\web.config
12:22:39 AM Downloading C:\inetpub\wwwroot\Microsoft.png
12:22:39 AM Downloading C:\inetpub\wwwroot\line.png
12:22:38 AM Downloading C:\inetpub\wwwroot\status.aspx
12:22:38 AM Downloading C:\inetpub\wwwroot\default.aspx
12:19:20 AM Installing Internet Information Server (this might take a few minutes)
12:19:16 AM Installing NuGet Package Provider
12:18:45 AM TemplateLink: https://raw.githubusercontent.com/Microsoft/nav-arm-templates/master/getnav.json
12:18:45 AM Starting initialization

<img width="760" alt="001" src="https://user-images.githubusercontent.com/26412023/41075300-a3ded73a-69d1-11e8-8a76-2d37007efad2.PNG">

**Comment by ajansari** [comment]
I should have just waited an extra few minutes before posting the comment. The Replace-NavServerContainer script just finished running and it does seem to have fixed the issue. The Web Client, Windows Client and Click Once install are all working with O365 credentials. 

Interestingly, the only thing that didn't update is the install status page. It still says stuck on the property 'Tenants' upgrade. I suspect the PowerShell script / cmdlet doesn't update that file. It's just a cosmetic issue, but I thought it would be worth mentioning.

TL/DR version: Still got error during scripted deployment, but Replace-NavServerContainer fixed the issue.

**Comment by freddydk** [comment]
What machine size do you select?

**Comment by AlphaGeeKUK** [comment]
I created a new machine this morning as well and get the same issue.

Machine size is Standard D2 v3 (2 vcpus, 8 GB memory)

Replace-NavServerContainer seems to have fix the issue except that ClickOnce client fails with a Sign in error. I've sent you an email with more info and logon credentials.

Same cosmetic thing on the landing page too "Installation NOT complete"

**Comment by freddydk** [comment]
and mine keeps working...
http://fk2018na.southcentralus.cloudapp.azure.com/status.aspx
even tried same datacenter as AJ.
Only difference now seems to be the O365 credentials, but mine is just freddyk at mynavdemo.onmicrosoft.com - no magic in that.

**Comment by AlphaGeeKUK** [comment]
I created another new VM using microsoft/dynamics-nav:2018-cu6-gb only this time I left the O365 parameters empty. The build completed ok.

Then I ran the Assisted Setup / Set up Azure Active Directory wizard in the Windows client, chose Auto-Register, login with the same credentials as I supplied to the template in the earlier builds and do not get a Key returned.

**Comment by ajansari** [comment]
I'm also selecting the default D2_v3.

Out of curiosity, Freddy, have you done anything manually in your Office 365 tenant or the AAD associated with the tenant to give certain permissions / admin privilege, etc.? The user I'm using is the administrator, but I'm wondering if the issue is indeed related to O365 and some privilege I need to extend to the user or in the tenant to allow the deployment script to run to completion.

The tenant I'm using was created from demos.microsoft.com where we get 90 day O365 E5 demos with 25 pre-created users and Dynamics 365 for Sales (Customer Engagement Plan). I do it so I can also easily set up the native integration between NAV and D365 Sales for presale demos.

**Comment by AlphaGeeKUK** [comment]
I have tried deploying a 2018 CU5 GB to see if it was a problem with CU6.

11:32:03 AM The property 'Tenants' cannot be found on this object. Verify that the property exists.
11:30:24 AM Creating Aad Apps for Office 365 integration

Looking more like an AAD and/or O365 issue.

I spun up a CU4 GB a few weeks ago and that worked ok.

**Comment by freddydk** [comment]
It is definitely a AAD issue - but it doesn't happen with my AAD creds - and it doesn't happen for you when you run replace-navservercontainer (in which you re-run the exact same code).
I probably need a set of failing AAD credentials to be able to troubleshoot further.

**Comment by ajansari** [comment]
In the interest of getting it fixed, we could do two things:

I'd be happy to "donate" my O365 creds if you'd like to use them to provision a new NAV on Docker instance from aka.ms/getnav. But, if you think these credentials won't give an error again because we've got them working now with my instance then....

I can whip up a new set of O365 demo creds from demos.microsoft.com (there are many choices for O365 demo tenants here, and I will create one using the same choices I used before), and I can e-mail them to you. And you can use them with aka.ms/getnav.

Thoughts?

**Comment by freddydk** [comment]
After a lot of testing, I have found a way to make this work.
Please retry http://aka.ms/getnav and give me a heads up if you still encounter issues.
Thanks for your help and your patience.

**Comment by AlphaGeeKUK** [comment]
Docker image: microsoft/dynamics-nav:2018-cu6-gb

Windows Client - OK
Web Client local - OK
Web Client Remote - OK (in an In Private Session due to all of my conflicting cookies)
Classic Dev - OK to edit Objects

Although the Installation Status page shows -
1:25:56 PM Creating Aad Apps for Office 365 integration
- I cannot see any App Registrations for the new server in the Azure Portal. Attempting to run the Assisted Setup's SETUP AZURE ACTIVE DIRECTORY wizard asks me to log in (which I do with a Global Admin account) but when the browser window closes, no App ID or Key is returned and the Status remains 'Not Complete' so I think the deployment script is not failing, but it's not doing anything either.

ClickOnce -Fails with Sign-in error -

```
**Sign in** 
Sorry, but we’re having trouble with signing you in.
AADSTS70001: Application with identifier 'https://tsgnav2018cu6.westeurope.cloudapp.azure.com/NAV/' was not found in the directory deca18fc-b56f-496f-bf7e-e28ca64ac076
```

The box at the side has this info -

```
Request Id:  df378bfe-1856-478c-b41c-9ea52a0d1a00 
Correlation Id:  b2f1a0c3-5a2b-461b-895e-a28fd8f001a9 
Timestamp:  2018-06-11T15:00:38Z 
Message:  AADSTS70001: Application with identifier 'https://tsgnav2018cu6.westeurope.cloudapp.azure.com/NAV/' was not found in the directory deca18fc-b56f-496f-bf7e-e28ca64ac076 
Advanced diagnostics: Enable 
If you plan on getting support for an issue, turn this on and try to reproduce the error. This will collect additional information that will help to troubleshoot the issue.

```
Which is similar to the problem I get if I don't try the Web Client in an In Private session (different 'was not found in directory' strings

```
**Sign in**
Sorry, but we’re having trouble with signing you in.

AADSTS70001: Application with identifier 'https://tsgnav2018cu6.westeurope.cloudapp.azure.com/NAV/' was not found in the directory 70f369b2-a94b-4465-8c6d-728adf25908d


```Box on the side -

```
Request Id: a11cb9b1-fead-4e33-81c2-f0f31c7b1500 
Correlation Id: de2c36e9-f700-497f-8afe-6791f557b814 
Timestamp: 2018-06-11T15:09:39Z 
Message: AADSTS70001: Application with identifier 'https://tsgnav2018cu6.westeurope.cloudapp.azure.com/NAV/' was not found in the directory 70f369b2-a94b-4465-8c6d-728adf25908d 
Advanced diagnostics: Enable 
If you plan on getting support for an issue, turn this on and try to reproduce the error. This will collect additional information that will help to troubleshoot the issue.

```
I tried to clear the cookies from Edge and IE, but the error with ClickOnce persists.

If I install the ClickOnce Win Client on a PC I've never logged on to any of Microsoft accounts on, ClickOnce works ok

 **What else can I try to fix this one? It's got to be something to do with cookies or some other stored settings on my local dev machine**

**Comment by freddydk** [comment]
In order to find the app registrations, you will have to login to portal.azure.com with the O365 credentials (NOT the credentials you have used to deploy the VM).
The other errors looks like you are signed in with a user from a different AAD as well (as you found when trying to login using a machine which have never been logged in before).
I am not sure how to get around the clickonce client logging you into the wrong account - that problem is unrelated to Docker.

**Comment by AlphaGeeKUK** [comment]
Hi Freddy,

I deployed the VM with the same O365 Global Admin account that I am using to login to login to the Azure portal - I am Admin in both...  I'm not sure what you mean.

Agree that ClickOnce is nothing to do with Docker.

**Comment by ajansari** [comment]
@freddydk, I'll give it a try too, and share my experience.

@AlphaGeeKUK, are the Web Client and Windows Client (non-ClickOnce) working fine with the O365 credentials, or are you having an issue with those as well?

**Comment by AlphaGeeKUK** [comment]
@ajansari The Web Client, Windows Client work ok logging into NAV with O365 credentials. The ClickOnce deployed Windows Client works ok too as long as I'm not on a machine like my dev box that has lots of other Microsoft accounts on it. I've got my Demo env (where I am building these machines), my live company and countless Office Demo accounts. I'm still trying to figure out how to make my dev box forget all of those login's saved credentials so ClickOnce deployed Windows Client will ask me to actually log in instead of using some credentials that are cached somewhere on my machine. But as Freddy says, this is not a Docker issue.

**Comment by freddydk** [comment]
I will close this issue as fixed.
